code/game/objects/structures/plasticflaps.dm

```diff
@@ -57,15 +57,15 @@
 	name = "airtight plastic flaps"
 	desc = "Heavy duty, airtight, plastic flaps."
 
-/obj/structure/plasticflaps/mining/New() //set the turf below the flaps to block air
-	var/turf/T = get_turf(loc)
-	if(T)
-		T.blocks_air = 1
-	..()
+/obj/structure/plasticflaps/mining/New()
+	air_update_turf(1)
+	. = ..()
+	
+/obj/structure/plasticflaps/mining/CanAtmosPass()
+	return FALSE
 
-/obj/structure/plasticflaps/mining/Destroy() //lazy hack to set the turf to allow air to pass if it's a simulated floor //wow this is terrible
-	var/turf/T = get_turf(loc)
-	if(T)
-		if(istype(T, /turf/open/floor))
-			T.blocks_air = 0
-	return ..()
+/obj/structure/plasticflaps/mining/Destroy()
+	var/atom/oldloc = loc
+	. = ..()
+	if (oldloc)
+		oldloc.air_update_turf(1)
```

Comment:
..() will set loc to null. the old loc will be the proper loc, it will work

anywho, i just fixed the pathing issue for air_update_turf: #19220 this change can wait for that pr to be merged.